### ISSUE TITLE
[Relay Mining] Implement Supplier driven RelayMiningDifficulty

### ISSUE THREAD

**Issue by red-0ne** [pokt-network/poktroll#818]
## Objective

Prevent malicious `Gateway`s from abusing on-chain `RelayMiningDifficulty` to get free `Relay`s by sending low enough volumes to `Supplier`s.

**Implement Supplier-Driven Relay Mining Difficulty**

Each `Claim` should specify the mining difficulty used to construct the `SparseMerkleSumTree`, which will be included in the `MsgCreateClaim` message.

During proof verification, the difficulty value should be retrieved from the respective `Claim` and used to verify whether the `Relay` satisfies the expected difficulty.

Additionally, this difficulty will be utilized to estimate the processing volume for the corresponding session.

## Origin Document

**Potential Exploit in On-Chain Relay Mining Difficulty: Unclaimable Relays**

The current on-chain `RelayMiningDifficulty` mechanism may be vulnerable to abuse, making `Suppliers` to perform unclaimable work. 

Malicious `Gateways` could exploit this by sending relays with volumes low enough to avoid triggering a relay mining process. By duplicating this process across multiple `Applications`, they could scale the exploit, leading to free work performed by `Suppliers`.

For more details, please refer to [RelayMiningDifficulty-and-low-volume](https://www.notion.so/buildwithgrove/RelayMiningDifficulty-and-low-volume-7aab3edf6f324786933af369c2fa5f01) Notion document.

## Goals

- Implement Supplier-driven RelayMiningDifficulty
- Disable (remove?) on-chain based RelayMiningDifficulty
- Ensure that very low volume (a few Relays) are still claimable.

## Deliverables

- [ ] A single PR that:
  - Implements Supplier driven RelayMiningDifficulty based on [RelayMiningDifficulty-and-low-volume#Algorithm](https://www.notion.so/buildwithgrove/RelayMiningDifficulty-and-low-volume-7aab3edf6f324786933af369c2fa5f01?pvs=4#7e1c196d19dd49cd9998f5fdf3d4ac94)
  - Verify Proofs using the RelayMiningDifficuly available in the corresponding claim.
  - Use the RelayMiningDifficuly available in the claim to calculate the estimated volume see #771 
  - Disable ServiceRelayMiningDifficulty (make sure they are not used).
  - Add min/max mined relays per claim parameters as constants.
  - Tests that ensure that low volume is claimable.

## Non-goals / Non-deliverables

- Create new governance parameters.
- Look for the right parameters to use.

## General deliverables

- [ ] **Comments**: Add/update TODOs and comments alongside the source code so it is easier to follow.
- [ ] **Testing**: Add new tests (unit and/or E2E) to the test suite.
- [ ] **Documentation**: Update architectural or development READMEs; use [mermaid](https://mermaid-js.github.io/mermaid/) diagrams where appropriate.

**Creator**: @adshmh 
**Co-Owners**: @bryanchriswhite, @red-0ne

**Comment by Olshansk** [comment]
@red-0ne Will review this in detail by EOD. Please hold off from starting work on it just yet

**Comment by Olshansk** [comment]
Copying this from a message on discord last week:


I just put up [1] for review as part of the ticket I'm handing off in [2] (related to proportional rewards), which is a pre-req for [3] that @red0ne put together earlier this week.

Note that I left a lot of TODO(arash) in the PR as a guideline for where/what needs to be implemented.

This is ULTRA critical for tokenomics to work properly.

[1] https://github.com/pokt-network/poktroll/pull/771
[2] https://github.com/pokt-network/poktroll/issues/781
[3] https://github.com/pokt-network/poktroll/issues/818

**Comment by Olshansk** [comment]
@adshmh I wanted to call out that we need to finish #771 before this can be started.

@red-0ne
1. I have updated the description here. PTAL and see if there are any updates/changes needed.
2. My only open question is related to this commend. What do you mean by it?
>   - Add min/max mined relays per claim parameters as constants.
3. Do you see other risks here where the supplier claims a high difficulty but actually inserted (in part) low difficulty relays into the tree? This could potentially be an attack as well, no?

**Comment by red-0ne** [comment]
@Olshansk,

min/max mined relays per claim are described here [1]

TL;DR;

**MinMinedRelaysPerClaim**: Since a `Supplier` is the one submitting the difficulty, it could submit a claim with only the most difficult relay thus leveraging any "luck" the supplier might have, which may not correspond to the actual work done.

**MaxMindedRelaysPerClaim**: On the other hand prevents the `Supplier` from submitting a claim with the easiest difficulty leading to long proof paths.

**Example**
With `min = 3` and `max = 5`
And the following served relays difficulty: `1, 2, 3, 4, 5, 6, 7, 8, 9, 20`

The supplier is **not allowed** to to submit the following claims:
* `{difficulty: 20, minedRelays: 1}`
* `{difficulty: 2, minedRelays: 9}`

But is allowed to submit:
* `{difficulty: 7, minedRelays: 4}`

[1] [RelayMiningDifficulty-and-low-volume#ProbabilisticError](https://www.notion.so/buildwithgrove/RelayMiningDifficulty-and-low-volume-7aab3edf6f324786933af369c2fa5f01?pvs=4#29f7511b57ae4c85864ab7ad671138f1)

**Comment by red-0ne** [comment]
Min/Max mined relays per claim are the equivalent of the `targetNumRelays` in the current setup

**Comment by Olshansk** [comment]
@red-0ne As discussed offline, do you think this is something we can do POST Mainnet?
